tempesta_fw/classifier/frang.c

```diff
@@ -956,8 +956,9 @@ frang_resp_code_range(const int n)
 static int
 frang_resp_handler(void *obj, struct sk_buff *skb, unsigned int off)
 {
-	TfwHttpReq *req = (TfwHttpReq *)obj;
-	TfwHttpResp *resp = (TfwHttpResp *)req->resp;
+	TfwConn *conn = (TfwConn *)obj;
+	TfwHttpResp *resp = container_of(conn->msg, TfwHttpResp, msg);
+	TfwHttpReq *req = TFW_HTTP_RESP_BOUND_REQ(resp);
```

Comment:
We can fix #838 by always keeping requests in a pair with responses and vise versa, as it's done here, or pass request and response along with L3/L4 data (skb, off) to GFSM callbacks. The design problem with GFSM is that it operates with skb and off, i.e. on L3/L4 layers only, while Frang requires HTTP requests and responses. So we definitelly need `TfwFsmData`. However, now probably we don't need both the `req` and `resp` in it and can replace it with the union. I think it's easier for a GFSM hook to get both the `req` and `resp` and check them for NULL rather than try to guess what's passed and get a pair. Opinions?